platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/DefaultSignedStateHasher.java

```diff
@@ -70,23 +70,27 @@ public DefaultSignedStateHasher(
      * {@inheritDoc}
      */
     @Override
-    public void hashState(@NonNull final StateAndRound stateAndRound) {
+    @Nullable
+    public StateAndRound hashState(@NonNull final StateAndRound stateAndRound) {
         final Instant start = Instant.now();
-        try {
+        try (final ReservedSignedState reservedSignedState = stateAndRound.reservedSignedState()) {
```

Comment:
Are you making an additional reservation here so that we do cleanup if we throw an exception?

Reply:
As we discussed, the purpose was for sake of each component in the pipeline getting a fresh reservation scoped with an appropriate string.